shardingsphere-agent/shardingsphere-agent-plugins/shardingsphere-agent-plugin-metrics/shardingsphere-agent-metrics-prometheus/src/main/java/org/apache/shardingsphere/agent/metrics/prometheus/wrapper/PrometheusWrapperFactory.java

```diff
@@ -0,0 +1,171 @@
+        Map metric = findById(id);
+        if (null == metric || null == getType(metric)) {
+            return Optional.ofNullable(null);
+        }
+        switch (getType(metric).toUpperCase()) {
+            case "COUNTER":
+                return createCounter(metric);
+            case "GAUGE":
+                return createGauge(metric);
+            case "HISTOGRAM":
+                return createHistogram(metric);
+            case "SUMMARY":
+                return createSummary(metric);
+            case "DELEGATE":
+                return Optional.of(new DelegateWrapper(id));
+            default:
+                return Optional.ofNullable(null);
+        }
+    }
+    
+    private Optional<MetricsWrapper> createCounter(final Map metric) {
+        Counter.Builder builder = Counter.build()
+                .name(getName(metric))
+                .help(getHelp(metric));
+        if (null != getLabels(metric)) {
+            builder.labelNames(getLabels(metric).stream().toArray(String[]::new));
+        }
+        Counter counter = builder.register();
+        CounterWrapper wrapper = new CounterWrapper(counter);
+        return Optional.of(wrapper);
+    }
+    
+    private Optional<MetricsWrapper> createGauge(final Map metric) {
+        Gauge.Builder builder = Gauge.build()
+                .name(getName(metric))
+                .help(getHelp(metric));
+        if (null != getLabels(metric)) {
+            builder.labelNames(getLabels(metric).stream().toArray(String[]::new));
+        }
+        Gauge gauge = builder.register();
+        GaugeWrapper wrapper = new GaugeWrapper(gauge);
+        return Optional.of(wrapper);
+    }
+    
+    private Optional<MetricsWrapper> createHistogram(final Map metric) {
+        Histogram.Builder builder = Histogram.build()
+                .name(getName(metric))
+                .help(getHelp(metric));
+        if (null != getLabels(metric)) {
+            builder.labelNames(getLabels(metric).stream().toArray(String[]::new));
+        }
+        if (null != getProps(metric)) {
+            parserHistogramProps(builder, getProps(metric));
+        }
+        Histogram histogram = builder.register();
+        HistogramWrapper wrapper = new HistogramWrapper(histogram);
+        return Optional.of(wrapper);
+    }
+    
+    private Optional<MetricsWrapper> createSummary(final Map metric) {
+        Summary.Builder builder = Summary.build()
+                .name(getName(metric))
+                .help(getHelp(metric));
+        if (null != getLabels(metric)) {
+            builder.labelNames(getLabels(metric).stream().toArray(String[]::new));
+        }
+        Summary summary = builder.register();
+        SummaryWrapper wrapper = new SummaryWrapper(summary);
+        return Optional.of(wrapper);
+    }
+    
+    private void parserHistogramProps(final Histogram.Builder builder, final Map props) {
+        if (null != props.get("buckets")) {
```

Comment:
Consider inverting the if to reduce indent.

Reply:
good